source/simulator/assembly.cc

```diff
@@ -1010,8 +1010,18 @@ namespace aspect
       computing_timer.enter_section ("   Assemble composition system");
 
     const unsigned int block_idx = advection_field.block_index(introspection);
+
+    if (!advection_field.is_temperature() && advection_field.compositional_variable!=0)
+      {
+        // allocate the system matrix for the current compositional field by taking the matrix
+        // from composition 0 (this is the place we allocate the matrix at)
+        const unsigned int block0_idx = AdvectionField::composition(0).block_index(introspection);
+        system_matrix.block(block_idx, block_idx).reinit(system_matrix.block(block0_idx, block0_idx));
```

Comment:
This creates a new matrix, but uses the same sparsity pattern as the copied-from block. can you make this clear in the comment above?